api/src/main/java/com/example/api/service/map/ChapterService.java

```diff
@@ -95,4 +96,35 @@ public void deleteChapter(Long chapterID) throws WrongUserTypeException, EntityN
         prevChapter.ifPresent(value -> value.setNextChapter(chapter.getNextChapter()));
         chapterRepo.deleteChapterById(chapterID);
     }
+
+    public void editChapter(EditChapterForm editChapterForm) throws RequestValidationException {
+        // user validation
+        String email = authService.getAuthentication().getName();
+        User professor = userRepo.findUserByEmail(email);
+        userValidator.validateProfessorAccount(professor, email);
+
+        // chapter validation for given editChapterForm
+        Chapter chapter = chapterRepo.findChapterById(editChapterForm.getChapterId());
+        chapterValidator.validateChapterIsNotNull(chapter, editChapterForm.getChapterId());
+        chapterValidator.validateChapterCreation(editChapterForm.getEditionForm());
+        chapterValidator.validateChapterEdition(editChapterForm.getEditionForm(), chapter);
+
+        ChapterForm chapterForm = editChapterForm.getEditionForm();
+
+        // edit basic chapter data
+        chapter.setName(chapterForm.getName());
+        chapter.setPosX(chapterForm.getPosX());
+        chapter.setPosY(chapterForm.getPosY());
+
+        // edit chapter activity map
+        ActivityMap chapterMap = chapter.getActivityMap();
+        chapterMap.setMapSizeX(chapterForm.getSizeX());
+        chapterMap.setMapSizeY(chapterForm.getSizeY());
+        chapterMap.setImage(fileRepo.findFileById(chapterForm.getImageId()));
+
+        activityMapService.saveActivityMap(chapterMap);
+
+        chapter.setActivityMap(chapterMap);
+        chapterRepo.save(chapter);
```

Comment:
Wydaje mi się, że jak mamy oznaczony serwis jako @Transactional to nie trzeba dawać save(), ale mogę się mylić

Reply:
Zakomentowałem linijki 128 i 125 i chyba zapisało, "u mnie działa". Dzięki, nie wiedziałem o tym ;)